pipeline-model-extensions/src/main/java/org/jenkinsci/plugins/pipeline/modeldefinition/withscript/WithScriptDescribable.java

```diff
@@ -50,10 +55,24 @@ public WithScriptScript getScript(CpsScript cpsScript) throws Exception {
         if (c == null)
             throw new IllegalStateException("Expected to be called from CpsThread");
 
-        return (WithScriptScript) cpsScript.getClass()
-                .getClassLoader()
-                .loadClass(getDescriptor().getScriptClass())
-                .getConstructor(CpsScript.class, this.getClass())
+        Class clz = null;
+        try {
+            clz = cpsScript.getClass().getClassLoader().loadClass(getDescriptor().getScriptClass());
+        } catch (ClassNotFoundException e) {
+            // This is special casing to deal with PluginFirstClassLoaders, which don't have a functional findResource method.
+            // That results in GroovyClassLoader.loadClass failing to find resources to parse and load.
+            PluginWrapper pw = Jenkins.getInstance().getPluginManager().whichPlugin(getDescriptor().getClass());
+            if (pw != null) {
+                URL res = pw.classLoader.getResource(getDescriptor().getScriptClass().replace('.', '/') + ".groovy");
+                if (res != null) {
+                    clz = ((GroovyClassLoader) cpsScript.getClass().getClassLoader()).parseClass(new GroovyCodeSource(res));
```

Comment:
This cannot return null, can it? So, assuming the simplification above to not bother with `PluginWrapper`, the `throw` should be in an `else` clause (`res == null`).